fix(oci): Use correct name referencing when tagging

craciunoiuc opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

craciunoiuc committed:
fix(oci): Remove unused internal attribute `name`

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Co-authored-by: Alexander Jung <alex@unikraft.io>

antoineco commented on oci/pack.go:
```diff
@@ -38,7 +38,7 @@ type ociPackage struct {
 
 	// Embedded attributes which represent target.Target
-	name    string
+	// name    string
```
If a name is no longer needed, let's remove the field.

nderjung committed:
fix(oci): Prevent recurse error by inputting correct string name

The `ociPackage` method `Name()` relies on the image reference attribute
being set, the value of which is set at this moment.  Correctly set the
input name by parsing the target's name when the `--oci-tag` flag is not
set.

Signed-off-by: Alexander Jung <alex@unikraft.io>

nderjung committed:
fix(oci): Correctly handle full reference names

This commit fixes two instances where a full reference to the image is
required in the format `name:tag` which are required when packaging and
fetching images locally.

Without this change, the `latest` tag is used by default which may not
be the case.  To accommodate the change a new standard `imageRef`
method is introduced to return this string value.

Signed-off-by: Alexander Jung <alex@unikraft.io>

antoineco commented on oci/pack.go:
```diff
@@ -389,7 +387,7 @@ func (ocipack *ociPackage) Pull(ctx context.Context, opts ...pack.PullOption) er
 	}
 
-	fullref := ocipack.Name()
+	fullref := fmt.Sprintf("%s:%s", ocipack.Name(), ocipack.Version())
```
nit: I see that this format string is used in two places. We might as well create a very small function to encapsulate this format to ensure that all  functions use the same.

nderjung replied:
I looked internally into [the `name` package](github.com/google/go-containerregistry/pkg/name) to see if we could re-use anything there but there was nothing from a first glance.  I've just done a fixup and will push the change now.

antoineco reviewed (approved):
Approving with a non-blocking comment.
